### PR TITLE
feat(project-access): introduce search for multiple Fiori artifacts

### PR DESCRIPTION
**Pull request by Klaus-Keller**
### Ticket: 
https://github.com/SAP/open-ux-tools/issues/1007

### Description
Introduce functionality to search folders, usually the workspace root folders, for Fiori artifacts like applications, adaptation projects, extension projects, and libraries.

### CODE REVIEW

File: packages/project-access/src/project/search.ts [hunk 1]
```diff
@@ -227,3 +263,145 @@ export async function findAllApps(wsFolders: WorkspaceFolder[] | string[] | unde
...
+    }
+    if (options.artifacts.includes('adaptations')) {
+        results.adaptations = await filterAdaptations(pathMap);
+    }
+    if (options.artifacts.includes('extensions')) {
+        results.extensions = await filterExtensions(pathMap);
+    }
+    if (options.artifacts.includes('libraries')) {
+        results.libraries = await filterLibraries(pathMap);
+    }
```
Comment (devinea) [inline packages/project-access/src/project/search.ts:405]: May want to run these searches in parallel using ` Promise.all()` or similar.

**Review by devinea — DISMISSED** [review]
Thanks @Klaus-Keller 
Code changes and tests look like they cover the require scenarios. 
I haven't tested locally. 
One nice to have suggestion

File: packages/project-access/src/project/search.ts (hunk 1 above)
Comment (Klaus-Keller) [reply packages/project-access/src/project/search.ts:405]: @devinea 
While I did this in some other places for HTTP request and could significantly gain performance by running requests in parallel, I've tried it for the search for apps by running the search in each workspace root parallel. I couldn't measure significant performance improvements, my assumption is that parallelized disk access doesn't really help as the disk is the single resources and access is sequential. Do you have different experience? I can easily change the code.

Comment (devinea) [reply packages/project-access/src/project/search.ts:405]: @Klaus-Keller I don't have experience on parallel disk access. It just looked like code that could run in parallel.
But if there is no obvious benefit then I am fine with it as is. Cheers

File: .changeset/purple-phones-yawn.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@sap-ux/project-access': minor
+---
+
+Add findFioriArtifact() to allow searching for apps, adaptation projects, extension projects, and libraries
```
Comment (slin-sap) [inline .changeset/purple-phones-yawn.md:5]: New API function name is findFioriArtifacts(). Missing a 's' in the cset.

**Review by slin-sap — CHANGES_REQUESTED** [review]

File: .changeset/purple-phones-yawn.md (hunk 2 above)
Comment (Klaus-Keller) [reply .changeset/purple-phones-yawn.md]: @slin-sap, thanks, fixed it.

**Review by devinea — APPROVED** [review]
Update lg

**Review by slin-sap — APPROVED** [review]
- Since new API is added, cset with minor version bump looks good.
- New API allows consumer to flexibly filter Fiori projects list based on project types. 
- New API function has tsdoc and unit test with clear usage instructions
- Change is backward compatible to filter Fiori apps.

**Pull request merged**